index.js

```diff
@@ -52,7 +52,13 @@ function extend(target, source) {
   for (const [key, value] of Object.entries(source)) {
     // recursively extend if target has the same key, otherwise just assign
     if (Object.prototype.hasOwnProperty.call(target, key)) {
-      extend(target[key], value);
+      if (key == '__compat') {
+        // If attempting to merge __compat, we have a double-entry
+        throw new Error(
```

Comment:
It might be nice to subclass `Error` (e.g., `DuplicateCompatError`) instead of throwing a generic `Error`.